pkg/nuget/lock/parse.go

```diff
@@ -49,7 +51,33 @@ func (p *Parser) Parse(r dio.ReadSeekerAt) ([]types.Library, []types.Dependency,
 				Version: packageContent.Resolved,
 			}
 			libs = append(libs, lib)
+
+			depId := utils.PackageID(packageName, packageContent.Resolved)
+			dependsOn := make([]string, 0)
+
+			for depName := range packageContent.Dependencies {
+				dependsOn = append(dependsOn, utils.PackageID(depName, targetContent[depName].Resolved))
+			}
+
+			if depsMap[depId] != nil {
+				dependsOn = append(dependsOn, depsMap[depId]...)
+				if dependsOn = utils.UniqueStrings(dependsOn); dependsOn == nil {
+					dependsOn = make([]string, 0)
+				}
+			}
```

Comment:
```suggestion
			if savedDependsOn, ok := depsMap[depId]; ok {
				dependsOn = utils.UniqueStrings(append(dependsOn, savedDependsOn...))
			}
```
we don't need to check for emptiness (see previous comment)